react/src/components/ResourceGroupSettingModal.tsx

```diff
@@ -0,0 +1,411 @@
+      mutation ResourceGroupSettingModalCreateMutation(
+        $name: String!
+        $input: CreateScalingGroupInput!
+      ) {
+        create_scaling_group(name: $name, props: $input) {
+          ok
+          msg
+        }
+      }
+    `);
+  const [commitAssociateDomain, isInFlightCommitAssociateDomain] =
+    useMutation<ResourceGroupSettingModalAssociateDomainMutation>(graphql`
+      mutation ResourceGroupSettingModalAssociateDomainMutation(
+        $domain: String!
+        $scaling_group: String!
+      ) {
+        associate_scaling_group_with_domain(
+          domain: $domain
+          scaling_group: $scaling_group
+        ) {
+          ok
+          msg
+        }
+      }
+    `);
+
+  const schedulerOpts = JSON.parse(resourceGroup?.scheduler_opts || '{}');
+
+  const INITIAL_FORM_VALUES = filterEmptyValues({
+    name: resourceGroup?.name,
+    description: resourceGroup?.description,
+    domain: currentDomain,
+    scheduler: resourceGroup?.scheduler ?? 'fifo',
+    allowedSessionTypes: schedulerOpts?.allowed_session_types ?? [
+      'batch',
+      'interactive',
+    ],
+    active: resourceGroup?.is_active ?? true,
+    public: resourceGroup?.is_public ?? true,
+    pendingTimeout: schedulerOpts?.pending_timeout
+      ? _.toNumber(schedulerOpts?.pending_timeout)
+      : null,
+    retriesToSkip: schedulerOpts?.config?.num_retries_to_skip
+      ? _.toNumber(schedulerOpts?.config?.num_retries_to_skip)
+      : null,
+    wsProxyAddress: resourceGroup?.wsproxy_addr,
+  });
+
+  return (
+    <BAIModal
+      destroyOnClose
+      title={
+        resourceGroup
+          ? t('resourceGroup.ModifyResourceGroup')
+          : t('resourceGroup.CreateResourceGroup')
+      }
+      onCancel={() => {
+        onRequestClose?.(false);
+      }}
+      okText={resourceGroup ? t('button.Update') : t('button.Create')}
+      okButtonProps={{
+        loading:
+          isInFlightCommitCreateResourceGroup ||
+          isInFlightCommitAssociateDomain ||
+          isInFlightCommitUpdateResourceGroup,
+      }}
+      onOk={() => {
+        formRef.current
+          ?.validateFields()
+          .then((values: FormInputType) => {
+            const input = {
+              driver: 'static',
+              scheduler: values.scheduler,
+              description: values.description ?? null,
+              driver_opts: '{}',
+              scheduler_opts: JSON.stringify({
+                allowed_session_types: values.allowedSessionTypes,
+                ...(values.pendingTimeout && {
+                  pending_timeout: values.pendingTimeout,
+                }),
+                ...(values.retriesToSkip && {
+                  config: {
+                    num_retries_to_skip: values.retriesToSkip,
+                  },
+                }),
+              }),
+              is_public: values.public,
+              is_active: values.active,
+              wsproxy_addr: values.wsProxyAddress,
```

Comment:
If you set the field name to snake_case, you don't need to rewrite input. It can be shorten like this.
```TS
const input = {...values, description: values.description ?? null, driver_opts: '{}', ...}
```